scripts/gitlog.py

```diff
@@ -0,0 +1,81 @@
+import argparse
+import json
+import os
+import pathlib
+import subprocess
+
+SCALE_BUILD_ROOT = pathlib.Path(__file__).parent.parent.resolve()
+
+
+def repo_json(dir, since):
+    print('Yellow')
+
+
+def git_command(dir, command):
+    _cmd = ['git', '-C', dir]
+    _cmd.extend(command)
+    # print(' '.join(_cmd))
+    cp = subprocess.run(_cmd, stdout=subprocess.PIPE, encoding='utf8')
+    output = []
+    for line in cp.stdout.split('\n'):
+        line = line.strip()
+        if line:
+            output.append(line)
+    return output
+
+
+def git_branch(dir):
+    return git_command(dir, ['branch', '--show-current'])[0]
+
+
+def git_origin(dir):
+    return git_command(dir, ['config', '--get', 'remote.origin.url'])[0]
+
+
+def git_commits(dir, since, long=False):
+    origin = git_origin(dir)
+    if origin.endswith('.git'):
+        origin = origin[:-4]
+    result = []
+    command = ['log', f'--since={since}', '--oneline', '--decorate=no']
+    if long:
+        command.append('--no-abbrev-commit')
+    for line in git_command(dir, command):
+        commit = line.split()[0]
+        data = {
+            'commit': commit,
+            'url': f'{origin}/commit/{commit}',
+            'text': line[len(commit) + 1:]
+        }
+        result.append(data)
+    return result
+
+
+def git_directories():
+    dirlist = [str(SCALE_BUILD_ROOT)]
```

Comment:
```suggestion
    return [SCALE_BUILD_ROOT.as_posix()] + [
        i.as_posix() for i in next(SCALE_BUILD_ROOT.iterdir(), None) if i and i.is_dir()
    ]
```

Reply:
Actually you can get rid of `git_directories` entirely and just do this in the `generate` function